src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -704,74 +704,100 @@ REDHAWK_PALEXPORT void PalPrintFatalError(const char* message)
     write(STDERR_FILENO, message, sizeof(message));
 }
 
+#ifdef __linux__
+size_t
+GetLogicalProcessorCacheSizeFromOS()
+{
+    size_t cacheSize = 0;
+
+#ifdef _SC_LEVEL1_DCACHE_SIZE
+    cacheSize = max(cacheSize, sysconf(_SC_LEVEL1_DCACHE_SIZE));
+#endif
+#ifdef _SC_LEVEL2_CACHE_SIZE
+    cacheSize = max(cacheSize, sysconf(_SC_LEVEL2_CACHE_SIZE));
+#endif
+#ifdef _SC_LEVEL3_CACHE_SIZE
+    cacheSize = max(cacheSize, sysconf(_SC_LEVEL3_CACHE_SIZE));
+#endif
+#ifdef _SC_LEVEL4_CACHE_SIZE
+    cacheSize = max(cacheSize, sysconf(_SC_LEVEL4_CACHE_SIZE));
+#endif
+    return cacheSize;
+}
+#endif
+
 bool QueryCacheSize()
 {
     bool success = true;
     g_cbLargestOnDieCache = 0;
 
 #ifdef __linux__
-    DIR* cpuDir = opendir("/sys/devices/system/cpu");
-    if (cpuDir == nullptr)
-    {
-        ASSERT_UNCONDITIONALLY("opendir on /sys/devices/system/cpu failed\n");
-        return false;
-    }
+    g_cbLargestOnDieCache = GetLogicalProcessorCacheSizeFromOS();
 
-    dirent* cpuEntry;
-    // Process entries starting with "cpu" (cpu0, cpu1, ...) in the directory
-    while (success && (cpuEntry = readdir(cpuDir)) != nullptr)
+    if (g_cbLargestOnDieCache == 0)
     {
-        if ((strncmp(cpuEntry->d_name, "cpu", 3) == 0) && isdigit(cpuEntry->d_name[3]))
+        DIR* cpuDir = opendir("/sys/devices/system/cpu");
```

Comment:
This additional logic theoretically enables getting cache size on non-intel processors (intel ones should be handled by the sysconf stuff). 
But I am not sure if the /sys/devices/system/cpu contains cache details on other architectures at all. On my Raspberry Pi 3, both with aarch64 and arm32 Linux distros, the cache details are not in there and I remember that @sdmaclea said that on his aarch64 Linux device, there were no cache details either.
@sergign60 are these present on your arm32 Linux distro / device?